src/bosh_azure_cpi/lib/cloud/azure/vms/vm_manager.rb

```diff
@@ -530,16 +530,19 @@ def _create_virtual_machine(instance_id, vm_params, network_interfaces, availabi
               @azure_client.delete_virtual_machine(resource_group_name, vm_name)
             end
 
-            if @use_managed_disks
-              os_disk_name = @disk_manager2.generate_os_disk_name(vm_name)
-              @disk_manager2.delete_disk(resource_group_name, os_disk_name)
-            else
-              storage_account_name = instance_id.storage_account_name
-              os_disk_name = @disk_manager.generate_os_disk_name(vm_name)
-              @disk_manager.delete_disk(storage_account_name, os_disk_name)
+            # os_disk and ephemeral_disk can be nil because of the new ephemeral_os_disk
+            unless vm_params[:os_disk].nil?
+              if @use_managed_disks
+                os_disk_name = @disk_manager2.generate_os_disk_name(vm_name)
+                @disk_manager2.delete_disk(resource_group_name, os_disk_name)
+              else
+                storage_account_name = instance_id.storage_account_name
+                os_disk_name = @disk_manager.generate_os_disk_name(vm_name)
+                @disk_manager.delete_disk(storage_account_name, os_disk_name)
 
-              # Cleanup invalid VM status file
-              @disk_manager.delete_vm_status_files(storage_account_name, vm_name)
+                # Cleanup invalid VM status file
+                @disk_manager.delete_vm_status_files(storage_account_name, vm_name)
+              end
```

Comment:
Don't we need to execute this also with the new `phemeral_os_disk` property or the disk is cleaned automatically in this case when something goes wrong?

Reply:
I only changed the indentation it's still in the same `else` block.
The logic will not be used for the new stuff because the `ephemeral_os_disk` is only supported with `managed_disks`